### PR TITLE
Update master-branch to Pastas v0.22.0

### PR DESCRIPTION
**Pull request by raoulcollenteur**
# Short Description

Dev Update master-branch to Pastas v0.22.0

# Checklist before PR can be merged:
- [ ] closes issue #xxxx
- [ ] is documented
- [ ] PEP8 compliant code
- [ ] tests added / passed
- [ ] Example Notebook (for new features)
- [ ] API changes documented in Release Notes

### CODE REVIEW

**Review by martinvonk — APPROVED** [review]
Looks good to go. I can't resolve the codacy trailing whitespaces though.

Maybe we can also update the [milestones](https://github.com/pastas/pastas/milestones) after this release?

File: pastas/model.py [hunk 1]
```diff
@@ -89,8 +89,11 @@ def __init__(self, oseries, constant=True, noisemodel=True, name=None,
             "tmin": None,
             "tmax": None,
             "freq": freq,
-            "warmup": (3650 * to_timedelta(freq) if freq[0].isdigit()
-                       else Timedelta(3650, freq)),
+            "warmup": (
```
Comment (dbrakenhoff) [inline pastas/model.py:92]: This still requires a look. Quick reminder for 1.0.

File: pastas/model.py [hunk 2]
```diff
@@ -889,7 +892,9 @@ def _get_sim_index(self, tmin, tmax, freq, warmup, update_sim_index=False):
                 break
 
         if self.sim_index is None or update_sim_index:
+            # TODO: sort out what to do for freq > "D"
```
Comment (dbrakenhoff) [inline pastas/model.py:895]: Same comment as above here, modeling on non-daily timestep can get tricky. Needs a look for 1.0

File: pastas/modelcompare.py [hunk 3]
```diff
@@ -0,0 +1,877 @@
...
+
+import matplotlib.pyplot as plt
+import numpy as np
+from pandas import DataFrame, concat
+
+import pastas as ps
+from warnings import warn
+
+
+class CompareModels:
```
Comment (dbrakenhoff) [inline pastas/modelcompare.py:11]: I like this new comparison class, but it's still a little confusing when you want to create your own comparison figure. Perhaps we can take a look for 1.0 to see if we can make this a bit more intuitive:
- Initializing your own axes mosaic
- Where do you pass the stressmodel dictionary, telling the comparison on which axes to model which contribution.
- Using .plot() for custom comparison figures?

File: pastas/noisemodels.py [hunk 4]
```diff
@@ -188,7 +177,8 @@ def weights(self, res, p):
 
         .. math:: w = 1 / sqrt((1 - exp(-2 \\Delta t / \\alpha)))
 
-        which are then normalized so that sum(w) = len(res)
+        which are then normalized so that sum(w) = len(res).
+Ò
```
Comment (dbrakenhoff) [inline pastas/noisemodels.py:181]: whats this?

File: pastas/stressmodels.py [hunk 5]
```diff
@@ -30,9 +30,9 @@
 
 logger = getLogger(__name__)
 
-__all__ = ["StressModel", "StressModel2", "Constant", "StepModel",
-           "LinearTrend", "RechargeModel", "WellModel", "TarsoModel",
-           "ChangeModel"]
+__all__ = ["StressModel", "Constant", "StepModel", "LinearTrend",
+           "RechargeModel", "WellModel", "TarsoModel", "ChangeModel",
+           "StressModel2"]
```
Comment (dbrakenhoff) [inline pastas/stressmodels.py:35]: still here?

**Review by dbrakenhoff — APPROVED** [review]

File: pastas/stressmodels.py (hunk 5 above)
Comment (dbrakenhoff) [reply pastas/stressmodels.py:35]: nvm, i see the deprecated class is still there one more version.

**Comment by dbrakenhoff** [comment]
I added to more minor fixes that I added locally but forgot to commit.

**Comment by raoulcollenteur** [comment]
Thanks for the reviews. Going to merge this PR now, and then we can work on the do-to-list with a clean slate :)

**Pull request merged**